### PR TITLE
Tweaks to resthook related endpoints

### PR DESCRIPTION
**Pull request by rowanseymour**
* Fixes some docstring mistakes
* Fixes some missing param filtering
* Adds some prefetches

Goes with https://github.com/rapidpro/rapidpro-python/pull/26

### CODE REVIEW

File: temba/api/v2/serializers.py [hunk 1]
```diff
@@ -678,7 +678,7 @@ class Meta:
 
 
 class ResthookSubscriberReadSerializer(ReadSerializer):
-    resthook = serializers.SlugField()
+    resthook = serializers.SerializerMethodField()
```
Comment (rowanseymour) [inline temba/api/v2/serializers.py]: SlugField doesn't actually use the method below but instead just calls unicode() on the object

File: temba/api/v2/serializers.py [hunk 2]
```diff
@@ -689,22 +689,18 @@ class Meta:
...
+    resthook = serializers.CharField(required=True)
+    target_url = serializers.URLField(required=True)
 
     def validate_resthook(self, value):
-        if value:
-            resthook = self.get_resthook(value)
-            if not resthook:
-                raise serializers.ValidationError("No resthook with slug: %s" % value)
-        return value
+        resthook = Resthook.objects.filter(is_active=True, org=self.context['org'], slug=value).first()
```
Comment (rowanseymour) [inline temba/api/v2/serializers.py]: Having the validate method return the resthook object saves fetching it three times

File: temba/api/v2/views.py [hunk 3]
```diff
@@ -1952,11 +1952,9 @@ class ResthookEndpoint(ListAPIMixin, BaseAPIView):
     model = Resthook
     serializer_class = ResthookReadSerializer
     pagination_class = ModifiedOnCursorPagination
-    throttle_scope = 'v2.api'
 
     def filter_queryset(self, queryset):
```
Comment (rowanseymour) [inline temba/api/v2/views.py]: BaseAPIView already provides a queryset filtered by org

File: temba/api/v2/views.py [hunk 4]
```diff
@@ -2038,29 +2036,40 @@ class ResthookSubscriberEndpoint(ListAPIMixin, CreateAPIMixin, DeleteAPIMixin, B
...
+     * **id** - the id of the resthook subscription you want to remove
 
     Example:
 
         DELETE /api/v2/resthook_subscribers.json?id=10404016
 
-    Response is status code 204 and an empty response
-
-        status code: 204
+    You will receive either a 204 response if a subscriber was deleted, or a 404 response if no matching subscriber was found.
```
Comment (rowanseymour) [inline temba/api/v2/views.py]: This is just same blurb from all the other docstrings

File: temba/api/v2/views.py [hunk 5]
```diff
@@ -2129,41 +2138,45 @@ class ResthookEventEndpoint(ListAPIMixin, BaseAPIView):
             {
                 "resthook": "new-report",
                 "data": {
-                    channel=105,
-                    flow=50505,
+                    "channel": 105,
```
Comment (rowanseymour) [inline temba/api/v2/views.py]: This should be valid JSON

File: temba/api/v2/views.py [hunk 6]
```diff
@@ -1952,11 +1952,9 @@ class ResthookEndpoint(ListAPIMixin, BaseAPIView):
     model = Resthook
     serializer_class = ResthookReadSerializer
     pagination_class = ModifiedOnCursorPagination
-    throttle_scope = 'v2.api'
```
Comment (rowanseymour) [inline temba/api/v2/views.py]: So far all v2 endpoints use the default 'v2' except the heavy contacts/messages/runs ones, so thinking resthook ones don't actually need their own scope

File: temba/api/v2/views.py [hunk 7]
```diff
@@ -1952,11 +1952,9 @@ class ResthookEndpoint(ListAPIMixin, BaseAPIView):
...
 
     def filter_queryset(self, queryset):
-        org = self.request.user.get_org()
-        return Resthook.objects.filter(org=org, is_active=True).order_by('-modified_on')
```
Comment (rowanseymour) [inline temba/api/v2/views.py]: Ordering is done by the paginator

**Comment by ericnewcomer** [comment]
Looks good.

**Pull request merged**